Automattic/simplenote-ios
# Hardened Security: Mark 2

### Notes:
We'll ship + merge a new Simperium release (when both, this PR and [Simperium's 587](https://github.com/Simperium/simperium-ios/pull/587) are good to go!!)

### Details:
In this PR we're implementing a `Password Reset Required` flow. During Authentication, whenever the password does not meet the new strenght requirements **and** the credentials are valid, the user will be presented with a **Password Reset Required** UI.

@aerych May I bug you with this one?
Thank you sir!!

Closes #745

### Test
1. Open simplenote.com and set a password that has less than 8 characters
2. Perform a fresh install
3. Attempt to Log Into your account
4. Verify the `Password Reset Required` Alert shows up
5. Verify that pressing **Cancel** dismisses the alert
6. Press `Log In` again
7. Verify that pressing **Reset** switches you over to Safari, and the Reset Form is onscreen
8. Verify that the Reset Form preloads your email, and that an email is sent when you're thru
9. Verify that pressing over the Reset Password email link (on your phone) opens a web that allows you to reset your PW
10. Verify that **when the PW reset flow is complete** you're taken back to Simplenote
11. Verify that whenever Simplenote comes back onScreen again the Password Field is reset
12. Verify that the (old / outdated) password is nuked

### Release
`RELEASE-NOTES.txt` was updated in f3c8991 with:

> Simplenote will request to upgrade insecure passwords whenever you log in

---

### Screenshots

<img width="200" src="https://user-images.githubusercontent.com/1195260/81851819-70b37780-9530-11ea-9997-cb0d26ff4926.png">
